src/macros/macro.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <QtCore>
+
+#include "proto/macro.pb.h"
+namespace proto = mixxx::track::io;
+
+const QLoggingCategory macroLoggingCategory("macros");
+
+/// A MacroAction is the building block of a Macro.
+/// It contains a position as well as the action to be taken at that position.
+///
+/// Note that currently only jumps to a target position are available, but that
+/// is subject to change.
+struct MacroAction {
+    MacroAction(){};
+    MacroAction(double position, double target)
+            : position(position), target(target){};
+    // use FramePos once https://github.com/mixxxdj/mixxx/pull/2961 is merged
+    double position;
+    double target;
+
+    bool operator==(const MacroAction& other) const {
+        return position == other.position && target == other.target;
+    }
+
+    inline bool operator!=(const MacroAction& other) const {
+        return !operator==(other);
+    }
+
+    proto::Macro_Action* serialize() const;
+    ;
+
+    enum Type : uint8_t {
+        JUMP = 0
+    };
+};
+
+/// A Macro stores a list of MacroActions as well as its current state and label.
+class Macro {
+  public:
+    static QByteArray serialize(const QVector<MacroAction>& actions);
+    static QVector<MacroAction> deserialize(const QByteArray& serialized);
+
+    Macro(bool enabled, bool loop, QString label, QVector<MacroAction> actions);
```

Comment:
Yeah it isn't implemented yet, just wanted to have some basics down - no need to perfect everything all ready, can we please ignore these flags for now :)

Reply:
I think this makes sense to use QFlags regardless. Then we don't need any references to loops in the code at all and you can easily add a new enum value later on without touching the function signature.

Reply:
Yes, I am planning to use QFlags here, but I don't wanna bother with that right now ;)

Reply:
added a todo, please resolve the conversation unless you want to add something important - I find todos in code much easier to keep track of rather than keeping review comments open